prboom2/src/dsda/features.h

```diff
@@ -86,12 +86,22 @@ typedef enum {
   // 63
 } dsda_feature_flag_t;
 
-#define FEATURE_SIZE 8
+#define BITMASK(b) (1 << ((b) % 8))
+#define BITSLOT(b) ((b) / 8)
+#define BITSET(a, b) ((a)[BITSLOT(b)] |= BITMASK(b))
+#define BITCLEAR(a, b) ((a)[BITSLOT(b)] &= ~BITMASK(b))
+#define BITTEST(a, b) ((a)[BITSLOT(b)] & BITMASK(b))
+#define BITNSLOTS(nb) ((nb + 8 - 1) / 8)
+
+#define FEATURE_SIZE 128
+#define FEATURE_SLOTS BITNSLOTS(FEATURE_SIZE)
+
+#define FOR_FEATURE_SLOT for (int f = 0; f < FEATURE_SLOTS; f++) {
+#define END_FEATURE_SLOT }
```

Comment:
Why do we need `FOR_FEATURE_SLOT` and `END_FEATURE_SLOT` macros? I think `for` loop is more readable.

Reply:
I think its easier to understand using macros since we use this loop so many times. But sure, made a new commit to remove the macros.